### PR TITLE
lwm2m: Add shell support for deleting object and resource instances

### PR DESCRIPTION
**Pull request by SeppoTakalo**
Extend the previous support of creating object instances by allowing creation of resource instances as well.
Similarly, add support for deleting.

### CODE REVIEW

**Review by rlubos — COMMENTED** [review]
Documentation (https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/connectivity/networking/api/lwm2m.rst#id59) should be updated as well. But otherwise looks good.

**Comment by SeppoTakalo** [comment]
Thanks. Updated documentation as well.

File: doc/connectivity/networking/api/lwm2m.rst [hunk 1]
```diff
@@ -717,7 +717,10 @@ required actions from the server side.
             -t   Write value as time_t
 
     create  :create PATH
-            Create object instance
+            Create object or resource instance
+
+    delete  :delete PATH
+            Create object or resource instance
```
Comment (rlubos) [inline doc/connectivity/networking/api/lwm2m.rst:723]:
```suggestion
            Delete object or resource instance
```

Comment (SeppoTakalo) [reply doc/connectivity/networking/api/lwm2m.rst]: Thanks. Fixed.

**Review by rlubos — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Pull request merged**